plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -172,16 +172,7 @@ public void renameNamespace(ConnectorSession session, String source, String targ
     public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> namespace)
     {
         SessionContext sessionContext = convert(session);
-        List<Namespace> namespaces;
-
-        if (namespace.isPresent() && namespaceExists(session, namespace.get())) {
-            namespaces = ImmutableList.of(Namespace.of(namespace.get()));
-        }
-        else {
-            namespaces = listNamespaces(session).stream()
-                    .map(Namespace::of)
-                    .collect(toImmutableList());
-        }
+        List<Namespace> namespaces = relevantNamespaces(session, namespace);
```

Comment:
> Fix namespace bug

make commit title a bit more detailed.

Reply:
it would be nice to have this commit as a separate PR for quicker merge

Reply:
https://github.com/trinodb/trino/pull/19980 for reference for others.